### PR TITLE
chore: add deprecation tag to 'to' prop on Link primitive

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

This PR will mark `to` prop of Link as deprecated

#### Description of how you validated changes

No functional changes but updated tests for `console.warn`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by lzhou0 — APPROVED** [review]

File: docs/src/pages/[platform]/components/link/react.mdx [hunk 1]
```diff
@@ -47,7 +47,7 @@ To create a Link which opens in a new tab, use the `isExternal` prop. Under the
 
 ### Routing Libraries
 
-To use the Link with React routing libraries, pass in routing components to the `as` prop. Below is an example using React Router:
+You can use a Link with any React routing libraries that support custom component. Below is an example using Link with React Router v5, in which custom navigation component is allowed by passing it through the `component` prop:
```
Comment (thaddmt) [inline docs/src/pages/[platform]/components/link/react.mdx:50]:
```suggestion
You can use a Link with any React routing library that supports custom components. Below is an example using Link with React Router v5, in which a custom navigation component is rendered by passing it through the `component` prop:
```

File: packages/react/src/primitives/Link/Link.tsx [hunk 2]
```diff
@@ -4,11 +4,16 @@ import classNames from 'classnames';
...
 import { View } from '../View';
+import { useDeprecationWarning } from '../../hooks/useDeprecationWarning';
 
 const LinkPrimitive: Primitive<LinkProps, 'a'> = (
   { as = 'a', children, className, isExternal, to, ...rest },
   ref
 ) => {
+  useDeprecationWarning({
+    shouldWarn: to != null,
+    message: `Please stop using 'to' prop as it will be deprecated soon.`,
```
Comment (thaddmt) [inline packages/react/src/primitives/Link/Link.tsx:15]: Should we suggest an alternative here?

File: packages/react/src/primitives/types/link.ts [hunk 3]
```diff
@@ -16,6 +16,8 @@ export interface LinkOptions {
   isExternal?: boolean;
 
   /**
+   * @deprecated
+   * Please stop using this porp as it will be deprecated soon
```
Comment (joebuono) [inline packages/react/src/primitives/types/link.ts:20]: nit typo: 'prop'

File: packages/react/src/primitives/Link/Link.tsx (hunk 2 above)
Comment (joebuono) [inline packages/react/src/primitives/Link/Link.tsx:15]: Just a suggestion: 

The Link component's `to` prop will soon be deprecated. Please see the Amplify UI documentation for using the Link component with routing libraries: https://ui.docs.amplify.aws/react/components/link#routing-libraries

File: packages/react/src/primitives/types/link.ts (hunk 3 above)
Comment (0618) [reply packages/react/src/primitives/types/link.ts:20]: Props table be auto-generated at build time, but we may still want to register it in the code base?

Comment (joebuono) [reply packages/react/src/primitives/types/link.ts:20]: Same suggestion as above:

The Link component's `to` prop will soon be deprecated. Please see the Amplify UI documentation for using the Link component with routing libraries: https://ui.docs.amplify.aws/react/components/link#routing-libraries

File: docs/src/pages/[platform]/components/link/react.mdx (hunk 1 above)
Comment (joebuono) [inline docs/src/pages/[platform]/components/link/react.mdx:50]: You can use a Link with any React routing library that supports custom components. Below is an example using Link with React Router v5, in which the Link is passed to the `component` prop as a custom navigation component.

**Review by joebuono — COMMENTED** [review]
Looks good! Just had a few minor word-smithing suggestions.

File: packages/react/src/primitives/Link/Link.tsx (hunk 2 above)
Comment (zchenwei) [reply packages/react/src/primitives/Link/Link.tsx:15]: Yeah, I can suggest people to use our `Link` as a child component of the Link from other routing libs, and use `to` prop on that Link instead

**Review by 0618 — APPROVED** [review]

**Review by joebuono — APPROVED** [review]

**Pull request merged**